rootio/wbuffer.go

```diff
@@ -153,19 +154,12 @@ func (w *WBuffer) WriteString(v string) { //FIXME :works only on small strings
 	if l < 255 {
 		b[0] = byte(l)
 		copy(b[1:], []byte(v))
-		var n int
-		n, w.err = w.w.Write(b)
-		w.w.c += n
+		w.w.Write(b)
 	} else {
 		b[0] = 255
-		copy(b[5:], []byte(v))
-		w.w.WriteByte(b[0]) //writing 255 in the first byte
-		w.w.c++
+		w.w.WriteByte(b[0])   //writing 255 in the first byte
 		w.WriteU32(uint32(l)) //writing the length as a uint32
-		w.w.c += 4
 		copy(b[5:], []byte(v))
-		var n int
-		n, w.err = w.w.Write(b[5:]) //writing the string
-		w.w.c += n
+		w.w.Write(b[5:]) //writing the string
```

Comment:
here you drop the error that is returned by `w.w.Write` on the floor.
you should store it inside `w.err`.